torch/_dynamo/trace_rules.py

```diff
@@ -0,0 +1,48 @@
+import enum
+import functools
+import importlib
+
+
+class TraceRule(enum.Enum):
```

Comment:
This need some docs.  I also liked @ezyang's idea of mapping it to context manager classes.  Which could simplify some other code.